client/src/test/java/io/getstream/chat/android/client/ChatClientImplTest.kt

```diff
@@ -0,0 +1,171 @@
+package io.getstream.chat.android.client
+
+import com.nhaarman.mockitokotlin2.mock
+import io.getstream.chat.android.client.api.ChatClientConfig
+import io.getstream.chat.android.client.events.ChatEvent
+import io.getstream.chat.android.client.events.ConnectedEvent
+import io.getstream.chat.android.client.events.DisconnectedEvent
+import io.getstream.chat.android.client.events.NewMessageEvent
+import io.getstream.chat.android.client.events.UnknownEvent
+import io.getstream.chat.android.client.logger.ChatLogLevel
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.models.EventType
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.notifications.handler.ChatNotificationHandler
+import io.getstream.chat.android.client.token.FakeTokenManager
+import io.getstream.chat.android.client.utils.observable.FakeChatSocket
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.util.Date
+
+internal class ChatClientImplTest {
+
+    companion object {
+        val eventA = ConnectedEvent(EventType.HEALTH_CHECK, Date(), User(), "")
+        val eventB = NewMessageEvent(EventType.MESSAGE_NEW, Date(), User(), "type:id", "type", "id", Message(), 0, 0, 0)
+        val eventC = DisconnectedEvent(EventType.CONNECTION_DISCONNECTED, Date())
+
+        val eventD = UnknownEvent("d", Date(), emptyMap<Any, Any>())
+        val eventE = UnknownEvent("e", Date(), mapOf<Any, Any>("cid" to "myCid"))
+        val eventF = UnknownEvent("f", Date(), emptyMap<Any, Any>())
+    }
+
+    lateinit var socket: FakeChatSocket
+    lateinit var client: ChatClient
+
+    lateinit var result: MutableList<ChatEvent>
+
+    @BeforeEach
+    fun setUp() {
+        val config = ChatClientConfig(
+            "api-key",
+            "hello.http",
+            "cdn.http",
+            "socket.url",
+            1000,
+            1000,
+            false,
+            ChatLogger.Config(ChatLogLevel.NOTHING, null),
+            ChatNotificationHandler(mock()),
+            FakeTokenManager("")
+        )
+
+        socket = FakeChatSocket()
+        client = ChatClientImpl(
+            config = config,
+            api = mock(),
+            socket = socket,
+            notifications = mock()
+        )
+        result = mutableListOf()
```

Comment:
you can inline this line with the declaration of result. The test always recreates itself from zero, no need to use setUp to instantiate all variables.

Reply:
This kind of initialization is here because in [stream-chat-android](https://github.com/GetStream/stream-chat-android/blob/50ae0246ff577af007a82771417e9b3262f048e5/library/src/test/resources/junit-platform.properties), we're using per-class test instances by default, and therefore need this kind of resetting between tests. I'll start a discussion about this, we should probably decide on one approach to use in all projects.

Reply:
BTW, Why do we need a shared instance for the result?

Reply:
We don't really need it to be shared, it's just extracted to the class level so that the tests have a consistent style, I suppose. I brought this setup over from the tests of the previous implementation.